### DISCUSSION TITLE
Strict mode

### DISCUSSION THREAD

**Discussion by gfoligna** [mivek/MetarParser#384] [Ideas]
Hi there!

I've been looking for any METAR parser online that comes with a strict mode toggle, I found this repo and the website, which are great, btw!
I'm part of a national course in Uruguay about meteorology observation, taught by the national meteorology institution entity ([INUMET](https://www.inumet.gub.uy/)). We are learning about coding METAR/SPECI messages and it would be really helpful to have an online system like yours to run in a strict mode (failing whenever there is ANY format issues: characters, more or less spaces than required, etc.).

As I do work in the IT industry since 2010 doing all kinds of infrastructure tasks, I'm excited that I could help on anything about this project of yours.

Best,
Gonzalo

**Comment by mivek** [comment]
Hello Gonzalo,

Thank you for your feedback.
I am not sure strict mode can be easily implemented, right now the parsing method uses regular expression to match patterns.
For example before parsing, I first split the message into tokens using this [regular expression ](https://github.com/mivek/MetarParser/blob/476cb6870c8885fd719a580dde2c4a8434cf70cc/metarParser-parsers/src/main/java/io/github/mivek/parser/AbstractWeatherCodeParser.java#L27). 
Can you provide an example of a METAR that should fail in strict mode?

**Comment by gfoligna** [comment]
Hi Mivek,

I'll take a look at that regex, and will see if I can suggest any other.

I'm thinking of double spaces or spaces between any character at the groups.
One I've noticed it could also fail at _strict mode_:
`SPECI SUSO 151125Z 10015G28KT 9999 CAVOK 30/22 Q1018`
Usually it is little details like that one.
